chevah_build

```diff
@@ -208,11 +204,44 @@ case $OS in
             # We favour the BSD-flavoured "install" over the default one.
             # "ar", "nm" and "ld" are included by default in the same path.
             export PATH=/usr/ccs/bin/:$PATH
+            # Solaris 10 has OpenSSL 0.9.7, which is unusable with cryptography.
+            export BUILD_CFFI="no"
+            PIP_LIBRARIES=""
+            EXTRA_LIBRARIES=$EXTRA_LIBRARIES_NO_CFFI
+        else
+            if [ ${ARCH} = "sparc" ]; then
+                # There's a problem with building readline in 11.2 for SPARC.
+                export BUILD_LIBEDIT="no"
+            fi
+            # Theoretically, cryptography 2.2.2 should work in Solaris 11.x, as
+            # it has newer OpenSSL, but EC crypto is enabled only in 11.3 SRU19.
+            if [ -f /usr/include/openssl/ecdh.h ]; then
+                # No "-Wconversion -Wno-error=sign-conversion" for Studio's cc.
+                PIP_LIBRARIES="\
+                    setproctitle==1.1.10 \
+                    cryptography==2.2.2chevah \
+                    pyOpenSSL==17.0.0 \
+                    scandir==1.7 \
+                    "
+            elif [ -f /usr/include/openssl/cmac.h ]; then
+                # 1.8.2 works with the crippled OpenSSL 1.0.1 in Solaris 11.2.
+                PIP_LIBRARIES="\
+                    setproctitle==1.1.10 \
+                    cryptography==1.8.2 \
+                    pyOpenSSL==17.0.0 \
+                    scandir==1.7 \
+                    "
+            else
+                # 1.6 still supported OpenSSL 1.0.0, the default in 11.0/11.1.
```

Comment:
Do we need to support all theses solaris 11 versions?

Reply:
My efforts were into having things work consistently, that's why I upgraded Jacu from 11.1 to 11.2. Obviously, we can't support all versions of Solaris 11, like we don't support building with GCC on Solaris. But I chose to document these conditions, because you never know when/if you are going to need them again… Or maybe someone else will need them, as this is a public repo.